File: aoc24/d9/solu.py

X = open("input.txt", "r").read()
X = "2333133121414131402"

# files = {0:int(X[0])}
# files.update({i-1:int(X[i]) for i in range(2,len(X), 2)})
files = [int(X[i]) for i in range(0,len(X), 2)]
space = [int(X[i]) for i in range(1,len(X), 2)]
print(space)


res = 0
L = len(files)
p = 0
k = 1
j = 0
i = 0
while i < len(space)-1:

    for fv in range(files[j]):
        res += j*p
        print(j,p, i, L-k)
        p += 1

    j += 1
    lv = L-k
    while space[i]:# and lv>i:
        files[lv] -= 1
        space[i] -= 1
        res += lv*p
        p += 1
        if space[i] == 0:
            if files[lv] == 0: 
                k += 1
                lv = L-k
            break
        if files[lv] == 0: 
            k += 1
            lv = L-k

    #if lv<i:
    #    break
    i += 1
    

print(res)


        

        

File: aoc23/d8/solu.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func readFileIntoArray(filename string) ([][]int, error) {
	file, err := os.Open(filename)
	if err != nil {
		fmt.Println("Error opening file:", err)
		return [][]int{}, err
	}
	defer file.Close() // Close the file when the function exits

	// Create a scanner to read the file line by line
	scanner := bufio.NewScanner(file)

	// Read and process each line
	var res = [][]int{}
	for scanner.Scan() {
		line := scanner.Text()
		sarr := strings.Split(line, "")

		var tmp = []int{}
		for _, i := range sarr {
			j, err := strconv.Atoi(i)
			if err != nil {
				panic(err)
			}
			tmp = append(tmp, j)
		}
		res = append(res, tmp)
	}

	if err := scanner.Err(); err != nil {
		fmt.Println("Error reading file:", err)
	}
	return res, err
}

func main() {
	filename := "input.txt"
	content, err := readFileIntoArray(filename)
	if err != nil {
		fmt.Println("Error reading file:", err)
		return
	}

	fmt.Println("File content:")
	fmt.Println(content[0])
}
